bundles/org.eclipse.swt/Eclipse SWT Browser/win32/org/eclipse/swt/browser/Edge.java

```diff
@@ -482,34 +576,39 @@ public void create(Composite parent, int style) {
 	browser.addListener(SWT.Move, this::browserMove);
 
 	containingEnvironment.instances().add(this);
+	// Sometimes when the shell of the browser is opened before the browser is
+	// initialized, nothing is drawn on the shell. We need browserResize to force
+	// the shell to draw itself again.
+	browserResize(new Event());
 }
 
 void browserDispose(Event event) {
 	containingEnvironment.instances.remove(this);
-
-	if (webView_2 != null) webView_2.Release();
-	if (environment2 != null) environment2.Release();
-	settings.Release();
-	webView.Release();
-	webView_2 = null;
-	environment2 = null;
-	settings = null;
-	webView = null;
-
-	// Bug in WebView2. Closing the controller from an event handler results
-	// in a crash. The fix is to delay the closure with asyncExec.
-	if (inCallback) {
-		ICoreWebView2Controller controller1 = controller;
-		controller.put_IsVisible(false);
-		browser.getDisplay().asyncExec(() -> {
-			controller1.Close();
-			controller1.Release();
-		});
-	} else {
-		controller.Close();
-		controller.Release();
-	}
-	controller = null;
+	webViewProvider.scheduleWebViewTask(() -> {
+		webViewProvider.getWebView(false).Release();
+		if (environment2 != null) environment2.Release();
+		if (settings != null) settings.Release();
+		if (!webViewProvider.isWebView_2Unavailable()) webViewProvider.getWebView_2(false).Release();
```

Comment:
Since many places check for `!...Unavailable()`. Wouldn't it be easier to have `...Available()`?

Reply:
For Available, I'll have to check isDone and !isCancelled
While for unavailable I just check isCcancelled
What seems better?

Reply:
If that was really necessary, it would be an indicator that the method would not do what you would expect it to do. That's basically the comment I made on the method result being meaningless when being called before the web view has been initialized. Since that now seems to be fixed by waiting for the initialization to finish in the `...Unavailable()` methods, shouldn't it be possible to just invert the evaluation of `isCancelled()`?

Reply:
yeah right